README.md

```diff
@@ -135,10 +135,50 @@ As you can see you can use custom DSL when checking the expected state.
  * The `RecordedEvent` itself is extended with dynamic properties so you can just use `event.time` or `event.bytesWritten` etc.
    This might be handy when you need an aggregation like this `jfrEvents['jdk.FileWrite']*.bytesWritten.sum() == expectedBytes`
 
+## Automatic Analysis
+
+
+Java Mission Control provides an Automatic Analysis tool that performs pattern analysis on recorded JFR events to determine common performance issues with Java applications.
+
+It is possible to write assertions against the Automatic Analysis results to verify that unit tests against common performance issues:
+
+```java
+import dev.morling.jfrunit.*;
+
+import static dev.morling.jfrunit.JfrEventsAssert.*;
+import static dev.morling.jfrunit.ExpectedEvent.*;
+
+@Test
+@EnableConfiguration("profile")
+public void automaticAnalysisTest() throws Exception {
+        System.gc();
+        Thread.sleep(1000);
+
+        jfrEvents.awaitEvents();
+
+        JfrAnalysisResults analysisResults = jfrEvents.automaticAnalysis();
+
+        //Inspect rules that fired
+        assertThat(analysisResults).contains(FullGcRule.class);
+        assertThat(analysisResults).doesNotContain(HeapDumpRule.class);
+
+        //Inspect severity of rule
+        assertThat(analysisResults).hasSeverity(FullGcRule.class, Severity.WARNING);
+
+        //Inspect score of rule
+        assertThat(analysisResults)
+        .contains(FullGcRule.class)
+        .scoresLessThan(80);
```

Comment:
What does that score mean?

Taking a step back, I'm still trying to wrap my head around the capabilities of the JMC rules. Are there other real-world examples coming to mind which would show the power of that feature?

Reply:
Each rule can generate a score between 0-100.  It roughly translates to the risk rating of a problem that has been detected. 0 being least likely to cause an issue to 100 being most likely to be problem.  The score implementation is specific to the Rule that is triggered.  

e.g. for the `ExceptionRule` it is a score based on number of exceptions per second : https://github.com/openjdk/jmc/blob/master/core/org.openjdk.jmc.flightrecorder.rules.jdk/src/main/java/org/openjdk/jmc/flightrecorder/rules/jdk/exceptions/ExceptionRule.java#L109

For the SockertWriteRule, it is score calculated from the maximum duration of all socket writes: https://github.com/openjdk/jmc/blob/master/core/org.openjdk.jmc.flightrecorder.rules.jdk/src/main/java/org/openjdk/jmc/flightrecorder/rules/jdk/io/SocketWriteRule.java#L135

This assertion allows users who are familiar with the Rules to assert that risk scores did not cross a certain threshold

Reply:
Ok, I see. Aren't we back to relying on characteristics of the environment then though? E.g. Exceptions per second might differ on the general performance and load of the machine running the test. So I'm wondering how stable/portable such test would be.

Reply:
Maybe they were poor examples, I think some of the rules will be dependent on environment and others not.  At present all the rules available from JMC are loaded.  Maybe I could curate a list of Rules that do not rely on environment and only load them?